### PR TITLE
[native_assets_cli] Add example using `dart_api_dl.h`

### PR DESCRIPTION
**Pull request by dcharkes**
Example showing how to use `dart_api_dl.h` in native assets.

Does **not** address yet:

* https://github.com/dart-lang/native/issues/839

### CODE REVIEW

**Comment by dcharkes** [comment]
@mosuem The health check worker fails with exit code 1 but no useful error message.

```
Run exit 1
  exit 1
  shell: /usr/bin/bash -e {0}
  env:
    FLUTTER_ROOT: /opt/hostedtoolcache/flutter/stable-[3](https://github.com/dart-lang/native/actions/runs/7194479308/job/19595090226?pr=858#step:14:3).16.3-x6[4](https://github.com/dart-lang/native/actions/runs/7194479308/job/19595090226?pr=858#step:14:4)
    PUB_CACHE: /opt/hostedtoolcache/flutter/stable-3.1[6](https://github.com/dart-lang/native/actions/runs/7194479308/job/19595090226?pr=858#step:14:6).3-x64/.pub-cache
Error: Process completed with exit code 1.
```

How do I figure out what's wrong?

**Comment by mosuem** [comment]
> @mosuem The health check worker fails with exit code 1 but no useful error message.
> How do I figure out what's wrong?

The log shows
```
Unhandled exception:
Connection closed before full header was received
#0      IOClient.send (package:http/src/io_client.dart:96:7)
<asynchronous suspension>
#1      GitHub.request (package:github/src/common/github.dart:413:30)
<asynchronous suspension>
#2      PaginationHelper.fetchStreamed (package:github/src/common/util/pagination.dart:38:20)
<asynchronous suspension>
#3      PaginationHelper.jsonObjects (package:github/src/common/util/pagination.dart:95:5)
<asynchronous suspension>
#4      _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:152:3)
<asynchronous suspension>
Error: Process completed with exit code 255.
```
So either a bug from `package:github` or the Github API.

**Comment by dcharkes** [comment]
> The log shows

How do I find the log from the GitHub UI?

File: pkgs/native_assets_cli/example/use_dart_api/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+An example that uses `dart_api_dl.h`.
```
Comment (mosuem) [inline pkgs/native_assets_cli/example/use_dart_api/README.md:1]: Expand this more, so people like me don't have to start looking what `dart_api_dl.h` is in the first place, and why one would want to use it.

**Review by mosuem — APPROVED** [review]

**Comment by dcharkes** [comment]
@mosuem Isn't the changelog entry detection somewhat strict?

**Pull request merged**